test/e2e/workload_test.go

```diff
@@ -0,0 +1,270 @@
+			f.EventuallyDeployment(deployment.ObjectMeta).Should(matcher.HaveInitContainer(util.StashInitContainer))
+
+			By("Waiting for restore to succeed")
+			f.EventuallyRestoreSessionPhase(restoreSession.ObjectMeta).Should(Equal(v1beta1.RestoreSessionSucceeded))
+
+			By("checking the workload data has been restored")
+			restoredData, err := f.ReadDataFromMountedDirectory(deployment.ObjectMeta, framework.GetPathsFromRestoreSession(&restoreSession))
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Compare between restore data and sample data")
+			Expect(restoredData).To(BeEquivalentTo(sampleData))
+
+		})
+
+		It("Should leader elect and Backup new deployment", func() {
+			deployment.Spec.Replicas = types.Int32P(2) // two replicas
+			By("Creating Deployment " + deployment.Name)
+			_, err = f.CreateDeployment(deployment)
+			Expect(err).NotTo(HaveOccurred())
+			err = util.WaitUntilDeploymentReady(f.KubeClient, deployment.ObjectMeta)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating sample data inside workload")
+			err = f.CreateSampleDataInsideWorkload(deployment.ObjectMeta, "test-data.txt")
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating storage Secret " + cred.Name)
+			err = f.CreateSecret(cred)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating new repository")
+			err = f.CreateRepository(repo)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating BackupConfiguration" + backupCfg.Name)
+			err = f.CreateBackupConfiguration(backupCfg)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for sidecar")
+			f.EventuallyDeployment(deployment.ObjectMeta).Should(matcher.HaveSidecar(util.StashContainer))
+
+			By("Waiting for leader election")
+			f.CheckLeaderElection(deployment.ObjectMeta, apis.KindDeployment, "backupconfig")
+
+			By("Waiting for BackupSession")
+			f.EventuallyBackupSessionCreated(backupCfg.ObjectMeta).Should(BeTrue())
+			bs := f.GetBackupSession(backupCfg.ObjectMeta)
+
+			By("Check for repository status updated")
+			f.EventuallyRepository(&deployment).Should(WithTransform(f.BackupCountInRepositoriesStatus, BeNumerically(">=", 1)))
+
+			By("Check for succeeded BackupSession")
+			f.EventuallyBackupSessionPhase(bs.ObjectMeta).Should(Equal(v1beta1.BackupSessionSucceeded))
+
+			By("Delete BackupConfiguration")
+			err = f.DeleteBackupConfiguration(backupCfg)
+
+			By("Waiting to remove sidecar")
+			f.EventuallyDeployment(deployment.ObjectMeta).ShouldNot(matcher.HaveSidecar(util.StashContainer))
+
+			By("Delete sample data inside workload")
+			err = f.CleanupSampleDataInsideWorkload(deployment.ObjectMeta, "test-data.txt")
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating Restore Session")
+			err = f.CreateRestoreSession(restoreSession)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Waiting for initContainer")
+			f.EventuallyDeployment(deployment.ObjectMeta).Should(matcher.HaveInitContainer(util.StashInitContainer))
+
+			By("Waiting for leader election")
```

Comment:
No need to check leader election for `Deployment` restore. Because, restore process can be completed before you identify leader and delete it.